src/extensions/replay/sessionrecording.ts

```diff
@@ -434,19 +433,31 @@ export class SessionRecording {
         this.sessionId = sessionId
     }
 
-    private _tryTakeFullSnapshot(): boolean {
-        // TODO this should ignore based on emit?
+    private _tryTakeFullSnapshot(withCustomEvent?: { tag: string; payload: any }): void {
         if (!this._captureStarted) {
-            return false
+            return
         }
-        try {
-            this.rrwebRecord?.takeFullSnapshot()
-            return true
-        } catch (e) {
-            // Sometimes a race can occur where the recorder is not fully started yet, so we can't take a full snapshot.
-            logger.error('Error taking full snapshot.', e)
-            return false
+        // try the same action three times 50 milliseconds apart
+        // if it fails all three times then we give up
+        // this is to account for the fact that the recorder might not be started yet
```

Comment:
Isn't there any state flag that we can read from `rrwebRecord` and check if the object is ready to accept events?

Reply:
they don't ... just the error when you try to use it 🙈